### PR TITLE
Fix some analyzer warnings

### PR DESCRIPTION
**Pull request by melotic**
This PR fixes some of the warnings from #202

* [x] CA1018
* [x] CA1032
* [x] CA1033
* [x] CA1052
* [x] CA1401
* [x] CA1508
* [x] CA1725
* [x] CA1725
* [x] CA1805
* [x] CA1813
* [x] CA1816
* [x] CA1822
* [x] CA1825
* [x] CA1835
* [x] CA1836
* [x] CA1838
* [x] CA2016
* [x] CA2101
* [x] CA2201
* [x] CA2208
* [x] CA2234
* [x] CA2249
* [x] CA5392
* [x] SA1200
* [x] VSTHRD200

### CODE REVIEW

File: src/Microsoft.ComponentDetection.Common/PathUtilityService.cs [hunk 1]
```diff
@@ -64,6 +64,7 @@ public bool IsRunningOnWindowsContainer
     /// <param name="output"> The pointer output. </param>
     /// <returns> A pointer <see cref= "IntPtr"/> to the absolute path of a file. </returns>
     [DllImport("libc", EntryPoint = "realpath")]
+    [DefaultDllImportSearchPaths(DllImportSearchPath.SafeDirectories)]
```
Comment (JamieMagee) [inline src/Microsoft.ComponentDetection.Common/PathUtilityService.cs:67]: How did you decide the search path here and below?

**Comment by JamieMagee** [comment]
Can you revert the commit for `CA1822`. It's going to massively conflict with #412

File: src/Microsoft.ComponentDetection.Common/PathUtilityService.cs (hunk 1 above)
Comment (melotic) [reply src/Microsoft.ComponentDetection.Common/PathUtilityService.cs:67]: There's no documentation of the `DllImportSearchPath` for Linux ([this search](https://github.com/search?q=repo%3Adotnet%2Fruntime%20SafeDirectories&type=code) is unhelpful as well). libc is in `/usr/lib` and I'd assume that is a "safe directory." I've searched other code that imports libc and nothing adds the `DefaultDllImportSearchPaths`. 

For `kernel32.dll` P/Invoke's, `kernel32.dll` should only be in `System32` (or perhaps `SysWoW64` on 32-bit applications). This is what [other MSFT Repos do](https://github.com/microsoft/CopyOnWrite/blob/64628b58124b357a7d9c7fed1cfda4bfa38ff654/lib/Windows/NativeMethods.cs#L122).

**Comment by melotic** [comment]
> Can you revert the commit for `CA1822`. It's going to massively conflict with #412

Done.

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**